Firestore/core/src/remote/bloom_filter.cc

```diff
@@ -115,30 +144,7 @@ bool BloomFilter::MightContain(absl::string_view value) const {
 
 bool operator==(const BloomFilter& lhs, const BloomFilter& rhs) {
   return lhs.bit_count() == rhs.bit_count() &&
-         lhs.hash_count() == rhs.hash_count() && CompareBits(lhs, rhs);
-}
-
-bool operator!=(const BloomFilter& lhs, const BloomFilter& rhs) {
-  return !(lhs == rhs);
-}
-
-bool CompareBits(const BloomFilter& lhs, const BloomFilter& rhs) {
-  if (lhs.bit_count() != rhs.bit_count()) {
-    return false;
-  }
-
-  for (int32_t i = 0; i < lhs.bit_count(); ++i) {
-    int32_t offset = i % 8;
-    const uint8_t& lhs_byte = lhs.bitmap()[i / 8];
-    const uint8_t& rhs_byte = rhs.bitmap()[i / 8];
-    const bool bit1 = (lhs_byte & (static_cast<uint8_t>(0x01) << offset));
-    const bool bit2 = (rhs_byte & (static_cast<uint8_t>(0x01) << offset));
-    if (bit1 != bit2) {
-      return false;
-    }
-  }
-
-  return true;
+         lhs.hash_count() == rhs.hash_count() && HasSameBits(lhs, rhs);
```

Comment:
nit: Remove the check for `lhs.bit_count() == rhs.bit_count()` because that is already taken into account in `HasSameBits()`.

Reply:
but, bloom filter has 3 attributes, even tho bit_count is checked inside HasSameBits, it looks nice to check all attributes in == operator.

Reply:
All right. If you feel strongly enough about this to write a comment then feel free to leave it as is :)

Reply:
not that strong